polus-infering-splinedist-plugin/src/utils.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import os 
+
+import numpy as np
+
+import bfio
+from bfio import BioReader
+
+from csbdeep.utils import normalize
+
+from splinedist.models import Config2D, SplineDist2D, SplineDistData2D
+from splinedist.utils import phi_generator, grid_generator
+
+import matplotlib
+matplotlib.rcParams["image.interpolation"] = None
+import matplotlib.pyplot as plt
+
+import filepattern
+from filepattern import FilePattern as fp
+
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("infer")
+logger.setLevel(logging.INFO)
+
+def create_plots(image,
+                 prediction,
+                 output_dir, 
+                 image_name):
+    
+    """ This function generates subplots of the 
+    original image with its predicted image.
+
+    Args:
+        input_len - the number of images in array_images and array_label
+        output_dir - the location where the jpeg files are saved
+        model - the neural network used to make the prediction
+    Returns:
+        None, saves images in output directory
+    """
+
+    fig, (a_image,a_prediction) = plt.subplots(1, 2, 
+                                               figsize=(8,5), 
+                                               gridspec_kw=dict(width_ratios=(1,1)))
+
+    plt_image = a_image.imshow(image)
+    a_image.set_title("Image")
+
+    plt_prediction = a_prediction.imshow(prediction)
+    a_prediction.set_title("Prediction")
+
+    plot_file = "{}.jpg".format(image_name)
+    plt.savefig(os.path.join(output_dir, plot_file))
+    plt.clf()
+    plt.cla()
+    plt.close(fig)
+
+
+def predict_nn(image_dir : str,
+                base_dir : str,
+                output_directory : str,
+                gpu : bool,
+                imagepattern : str,):
+    """ The function uses the model saved in base_dir to predict the 
+    segmentations of the images in the image_dir.
+
+    Args:
+    image_dir: Directory with all the images 
+    base_dir: Directory of the model's weights
+    output_directory: Directory where the outputs gets saved
+    gpu: Specifies whether or not there is a GPU to use
+    imagepattern: Pattern of the files the user wants to use
+
+    Returns:
+    None, an output_directory filled with jpeg images of segmentations
+        and the performance of the neural network on every image
+
+    Raises:
+    Assertion Error: 
+        If the base_dir does not exist
+        If the phi_file does not exist
+        If the grid_file does not exist
+    """
+
+    assert os.path.exists(base_dir), \
+        "{} does not exist".format(base_dir)
+
+    fp_images = fp(image_dir,imagepattern)
+    images = []
+    for files in fp_images():
+        image = files[0]['file']
+        if os.path.exists(image):
+            images.append(image)
```

Comment:
Why check for file existence? If the file is grabbed by `filepattern`, then it should exist.